src/qt/transactionrecord.cpp

```diff
@@ -257,7 +261,21 @@ void TransactionRecord::updateStatus(const CWalletTx &wtx)
 bool TransactionRecord::statusUpdateNeeded()
 {
     AssertLockHeld(cs_main);
-    return status.cur_num_blocks != chainActive.Height();
+    {
+        // to detect mempool conflicts, we need to check each
+        // transaction if the mempool status has been changed.
+        // Could be further improved by a signal fired when a
+        // mempool transaction has been evicted
+
+        LOCK(mempool.cs);
+        if (status.inMempool != mempool.exists(hash))
+            return true;
+    }
+
+    if (status.cur_num_blocks != chainActive.Height())
```

Comment:
Nit: I think you can leave those three lines the way it was before: `return status.cur_num_blocks != chainActive.Height();`